### ISSUE TITLE
Infer(v0.13.1) installation failure on ubuntu 16.04 LTS

### ISSUE THREAD

**Issue by pnt4kor** [facebook/infer#914]
Hello,

I am beginner to the infer tool. Now I am trying to do some POCs using infer tool.
But, I am not able to succeed inferv0.13.1 installation on ubuntu 16.04 machine.  

Infer source : 
Steps followed for infer installation
1) install the dependencies 
`sudo apt-get install autoconf automake cmake curl git libc6-dev libsqlite3-dev opam openjdk-8-jdk-headless pkg-config python2.7 zlib1g-dev`
2) Downloaded the infer source
https://github.com/facebook/infer/releases/download/v0.13.1/infer-linux64-v0.13.1.tar.xz
3) Launch ./build-infer.sh

console output
```
[ERROR] The compilation of core failed at "jbuilder build -p core -j 4".

#=== ERROR while installing core.v0.10.0 ======================================#
# opam-version         1.2.2
# os                   linux
# command              jbuilder build -p core -j 4
# path                 /home/ubuntu/.opam/4.05.0+flambda/build/core.v0.10.0
# compiler             4.05.0+flambda
# exit-code            1
# env-file             /home/ubuntu/.opam/4.05.0+flambda/build/core.v0.10.0/core-13759-5f1023.env
# stdout-file          /home/ubuntu/.opam/4.05.0+flambda/build/core.v0.10.0/core-13759-5f1023.out
# stderr-file          /home/ubuntu/.opam/4.05.0+flambda/build/core.v0.10.0/core-13759-5f1023.err
### stderr ###
# [...]
#       ocamlc src/core__Lock_file.{cmo,cmt}
#       ocamlc src/core__Uuid.{cmo,cmt}
#       ocamlc src/core__Schedule_v4_deprecated.{cmi,cmti}
#       ocamlc src/core__Mac_address.{cmi,cmti}
#       ocamlc src/core__Time_ns_intf.{cmi,cmo,cmt}
#       ocamlc src/core__Version_util.{cmi,cmti}
#     ocamlopt src/core__Core_unix.{cmx,o} (got signal KILL)
# (cd _build/default && /home/ubuntu/.opam/4.05.0+flambda/bin/ocamlopt.opt -w -40 -safe-string -g -I /home/ubuntu/.opam/4.05.0+flambda/lib/base -I /home/ubuntu/.opam/4.05.0+flambda/lib/base/caml -I /home/ubuntu/.opam/4.05.0+flambda/lib/base/md5 -I /home/ubuntu/.opam/4.05.0+flambda/lib/base/shadow_stdlib -I /home/ubuntu/.opam/4.05.0+flambda/lib/bin_prot -I /home/ubuntu/.opam/4.05.0+flambda/lib/bin_prot/shape -I /home/ubuntu/.opam/4.05.0+flambda/lib/core_kernel -I /home/ubuntu/.opam/4.05.0+flambda/lib/core_kernel/base_for_tests -I /home/ubuntu/.opam/4.05.0+flambda/lib/fieldslib -I /home/ubuntu/.opam/4.05.0+flambda/lib/jane-street-headers -I /home/ubuntu/.opam/4.05.0+flambda/lib/ocaml -I /home/ubuntu/.opam/4.05.0+flambda/lib/ocaml/threads -I /home/ubuntu/.opam/4.05.0+flambda/lib/ppx_assert/runtime-lib -I /home/ubuntu/.opam/4.05.0+flambda/lib/ppx_bench/runtime-lib -I /home/ubuntu/.opam/4.05.0+flambda/lib/ppx_compare/runtime-lib -I /home/ubuntu/.opam/4.05.0+flambda/lib/ppx_expect/collector -I /home/ubuntu/.opam/4.05.0+flambda/lib/ppx_expect/common -I /home/ubuntu/.opam/4.05.0+flambda/lib/ppx_expect/config -I /home/ubuntu/.opam/4.05.0+flambda/lib/ppx_hash/runtime-lib -I /home/ubuntu/.opam/4.05.0+flambda/lib/ppx_inline_test/config -I /home/ubuntu/.opam/4.05.0+flambda/lib/ppx_inline_test/runtime-lib -I /home/ubuntu/.opam/4.05.0+flambda/lib/sexplib -I /home/ubuntu/.opam/4.05.0+flambda/lib/sexplib/0 -I /home/ubuntu/.opam/4.05.0+flambda/lib/sexplib/unix -I /home/ubuntu/.opam/4.05.0+flambda/lib/spawn -I /home/ubuntu/.opam/4.05.0+flambda/lib/stdio -I /home/ubuntu/.opam/4.05.0+flambda/lib/typerep -I /home/ubuntu/.opam/4.05.0+flambda/lib/variantslib -no-alias-deps -I src -open Core__ -o src/core__Core_unix.cmx -c -impl src/core_unix.pp.ml)
#
# Waiting for 3 jobs to finish.
```

- Is infer installation supported in Centos and RHEL? 

- Docker image creation was also not successful. System waits for the long time after installing opam lock 1.2
Docker log
```
=-=- Processing actions -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
-> installed lock.1.2
Done.
[WARNING] Running as root is not recommended
[ERROR] The compilation of core failed at "jbuilder build -p core -j 1".

#=== ERROR while installing core.v0.10.0 ======================================#
# opam-version         1.2.2
# os                   linux
# command              jbuilder build -p core -j 1
# path                 /root/.opam/4.05.0+flambda/build/core.v0.10.0
# compiler             4.05.0+flambda
# exit-code            1
# env-file             /root/.opam/4.05.0+flambda/build/core.v0.10.0/core-6648-405960.env
# stdout-file          /root/.opam/4.05.0+flambda/build/core.v0.10.0/core-6648-405960.out
# stderr-file          /root/.opam/4.05.0+flambda/build/core.v0.10.0/core-6648-405960.err
### stderr ###
# md5_stubs.c:10:32: warning: '-pedantic' is not an option that controls warnings [-Wpragmas]
# [...]
#     ocamlopt src/core__Core_thread.{cmx,o}
#       ocamlc src/core__Syscall_result.{cmo,cmt}
#     ocamlopt src/core__Syscall_result.{cmx,o}
#       ocamlc src/core__Core_unix.{cmi,cmti}
#       ocamlc src/core__Core_unix.{cmo,cmt}
#     ocamlopt src/core__Core_unix.{cmx,o} (exit 2)
# (cd _build/default && /root/.opam/4.05.0+flambda/bin/ocamlopt.opt -w -40 -safe-string -g -I /root/.opam/4.05.0+flambda/lib/base -I /root/.opam/4.05.0+flambda/lib/base/caml -I /root/.opam/4.05.0+flambda/lib/base/md5 -I /root/.opam/4.05.0+flambda/lib/base/shadow_stdlib -I /root/.opam/4.05.0+flambda/lib/bin_prot -I /root/.opam/4.05.0+flambda/lib/bin_prot/shape -I /root/.opam/4.05.0+flambda/lib/core_kernel -I /root/.opam/4.05.0+flambda/lib/core_kernel/base_for_tests -I /root/.opam/4.05.0+flambda/lib/fieldslib -I /root/.opam/4.05.0+flambda/lib/jane-street-headers -I /root/.opam/4.05.0+flambda/lib/ocaml -I /root/.opam/4.05.0+flambda/lib/ocaml/threads -I /root/.opam/4.05.0+flambda/lib/ppx_assert/runtime-lib -I /root/.opam/4.05.0+flambda/lib/ppx_bench/runtime-lib -I /root/.opam/4.05.0+flambda/lib/ppx_compare/runtime-lib -I /root/.opam/4.05.0+flambda/lib/ppx_expect/collector -I /root/.opam/4.05.0+flambda/lib/ppx_expect/common -I /root/.opam/4.05.0+flambda/lib/ppx_expect/config -I /root/.opam/4.05.0+flambda/lib/ppx_hash/runtime-lib -I /root/.opam/4.05.0+flambda/lib/ppx_inline_test/config -I /root/.opam/4.05.0+flambda/lib/ppx_inline_test/runtime-lib -I /root/.opam/4.05.0+flambda/lib/sexplib -I /root/.opam/4.05.0+flambda/lib/sexplib/0 -I /root/.opam/4.05.0+flambda/lib/sexplib/unix -I /root/.opam/4.05.0+flambda/lib/spawn -I /root/.opam/4.05.0+flambda/lib/stdio -I /root/.opam/4.05.0+flambda/lib/typerep -I /root/.opam/4.05.0+flambda/lib/variantslib -no-alias-deps -I src -open Core__ -o src/core__Core_unix.cmx -c -impl src/core_unix.pp.ml)
# File "src/core_unix.ml", line 1:
# Error: I/O error: as -o 'src/core__Core_unix.o' '/tmp/camlasmde0b16.s': Cannot allocate memory


The former state can be restored with:
    opam switch import
"~/.opam/4.05.0+flambda/backup/state-20180408090453.export"
Exception:
Reqs_error
 (Process_error
   ("opam install -y ANSITerminal.0.8 atd.1.12.0 atdgen.1.12.0 base.v0.10.0 base64.2.2.0 bin_prot.v0.10.0 biniou.1.2.0 camlp4.4.05+1 camlzip.1.07 cmdliner.1.0.2 conf-aclocal.1.0.0 conf-autoconf.0.1 conf-m4.1 conf-pkg-config.1.0 conf-which.1 configurator.v0.10.0 core.v0.10.0 core_kernel.v0.10.0 cppo.1.6.0 cppo_ocamlbuild.1.6.0 ctypes.0.13.1 easy-format.1.3.0 extlib-compat.1.7.2 fieldslib.v0.10.0 integers.0.2.2 jane-street-headers.v0.10.0 javalib.2.3.4 jbuilder.1.0+beta16 menhir.20171206 mtime.1.1.0 num.0 ocaml-compiler-libs.v0.10.0 ocaml-migrate-parsetree.1.0.7 ocamlbuild.0.12.0 ocamlfind.1.7.3 octavius.1.2.0 ounit.2.0.5 parmap.1.0-rc8 ppx_assert.v0.10.0 ppx_ast.v0.10.0 ppx_base.v0.10.0 ppx_bench.v0.10.0 ppx_bin_prot.v0.10.0 ppx_compare.v0.10.0 ppx_core.v0.10.0 ppx_custom_printf.v0.10.0 ppx_derivers.1.0 ppx_deriving.4.2.1 ppx_driver.v0.10.1 ppx_enumerate.v0.10.0 ppx_expect.v0.10.0 ppx_fail.v0.10.0 ppx_fields_conv.v0.10.0 ppx_hash.v0.10.0 ppx_here.v0.10.0 ppx_inline_test.v0.10.0 ppx_jane.v0.10.0 ppx_js_style.v0.10.0 ppx_let.v0.10.0 ppx_metaquot.v0.10.0 ppx_optcomp.v0.10.0 ppx_optional.v0.10.0 ppx_pipebang.v0.10.0 ppx_sexp_conv.v0.10.0 ppx_sexp_message.v0.10.0 ppx_sexp_value.v0.10.0 ppx_tools.5.0+4.05.0 ppx_traverse.v0.10.0 ppx_traverse_builtins.v0.10.0 ppx_type_conv.v0.10.0 ppx_typerep_conv.v0.10.0 ppx_variants_conv.v0.10.0 re.1.7.1 result.1.2 sawja.1.5.2 sexplib.v0.10.0 spawn.v0.10.1 sqlite3.4.3.2 stdio.v0.10.0 topkg.0.9.1 typerep.v0.10.0 variantslib.v0.10.0 xmlm.1.3.0 yojson.1.4.0 --color=never",
   Unix.WEXITED 4)).

*** `install_opam_deps` failed
*** Updating opam then retrying
```

Can you please check and provide your valuable input. Thanks in advance!

Regards,
Prashanth

**Comment by jvillard** [comment]
(updated your comment to use triple-backticks around pre-formatted code for readability)

Looks like your system ran out of memory trying to compile an OCaml dependency (Core) in both cases. Can you try infer 0.14.0 instead?

**Comment by TomMD** [comment]
N.B. I just built infer HEAD the other day and observed a peak disk space use of ~65 GB.

**Comment by pnt4kor** [comment]
@jvillard : Thank you! This solved the problem of infer installation on UBUNTU 16.04.  The reason is out of memory issue :)

But now I am facing issue in installing infer on RHEL. Can you please provide installation guide for it?
I tried installing opam package manually using
`wget https://raw.github.com/ocaml/opam/master/shell/opam_installer.sh -O - | sh -s /usr/bin`
then Launched `./build-infer.sh`
Log : 
```
=-=- Installing compiler 4.06.1+flambda -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[default.comp] https://github.com/ocaml/ocaml/archive/4.06.1.tar.gz downloaded
Now compiling OCaml. This may take a while, please bear with us...
Done.

=-=- Gathering sources =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=

=-=- Processing actions -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
∗  installed base-bigarray.base
∗  installed base-threads.base
∗  installed base-unix.base
Done.
# To setup the new switch in the current shell, you need to run:
eval `opam config env`

installing infer dependencies; this can take up to 30 minutes...
opam-lock not found in the current switch, installing from 'https://github.com/rgrinberg/opam-lock'...
Package lock does not exist, create as a NEW package ? [Y/n] y
lock is now git-pinned to https://github.com/rgrinberg/opam-lock

[lock] https://github.com/rgrinberg/opam-lock updated
[lock] Installing new package description from https://github.com/rgrinberg/opam-lock

lock needs to be installed.
The following actions will be performed:
  ∗  install lock 1.2*

=-=- Gathering sources =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[lock] https://github.com/rgrinberg/opam-lock already up-to-date

=-=- Processing actions -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
∗  installed lock.1.2
Done.
[ERROR] The compilation of camlzip failed at "make all".
[ERROR] The compilation of sqlite3 failed at "jbuilder build -p sqlite3 -j 4".
^C[ERROR] User interruption

[ec2-user@ip-172-31-8-167 infer-linux64-v0.14.0]$
The former state can be restored with:
    opam switch import
"~/.opam/infer-4.06.1+flambda/backup/state-20180411052912.export"
Fatal error: exception Sys_error("Broken pipe")
^C
[ec2-user@ip-172-31-8-167 infer-linux64-v0.14.0]$ sqlite3 --version
3.7.17 2013-05-20 00:56:22 118a3b35693b134d56ebd780123b7fd6f1497668
[ec2-user@ip-172-31-8-167 infer-linux64-v0.14.0]$ which sqlite3
/usr/bin/sqlite3
```
Thanks in advance!

Regards,
Prashanth

**Comment by jvillard** [comment]
You can get more precise error messages with `eval $(opam config env); opam install camlzip sqlite3`. I suspect you are missing the development libraries for zlib and sqlite3. Look for packages with names like `zlib-devel` and `libsqlite3-devel`?

**Comment by jvillard** [comment]
I'm assuming this has been solved, or at least that 0.15.0 removes the issue of having to compile infer, but if not feel free to reopen with more details.
